src/adapter/graphql/tests/tests/test_gql_dataset_flow_configs.rs

```diff
@@ -0,0 +1,730 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::sync::Arc;
+
+use async_graphql::{value, Value};
+use dill::Component;
+use event_bus::EventBus;
+use indoc::indoc;
+use kamu::testing::MetadataFactory;
+use kamu::{DatasetRepositoryLocalFs, DependencyGraphServiceInMemory};
+use kamu_core::{auth, CreateDatasetResult, DatasetRepository, SystemTimeSourceDefault};
+use kamu_flow_system_inmem::{FlowConfigurationEventStoreInMem, FlowConfigurationServiceInMemory};
+use opendatafabric::*;
+
+use crate::utils::{authentication_catalogs, expect_anonymous_access_error};
+
+////////////////////////////////////////////////////////////////////////////////////////
+
+#[test_log::test(tokio::test)]
+async fn test_crud_time_delta_root_dataset() {
+    let harness = FlowConfigHarness::new();
+
+    let create_result = harness.create_root_dataset().await;
+
+    let request_code = indoc!(
+        r#"
+        {
+            datasets {
+                byId (datasetId: "<id>") {
+                    flowConfigs {
+                        flowConfig (datasetFlowType: "INGEST") {
```

Comment:
Perhaps a structure like this:
```
datasets {
  byId(...) {
    flows {
      configs {
        byType(...) {
```
would be more consistent with our current API style

Reply:
Restructured, although it restored another level with configs.